tools/hermes-parser/js/flow-api-translator/src/flowToFlowDef.js

```diff
@@ -986,23 +986,43 @@ function convertSuperClass(
     return EMPTY_TRANSLATION_RESULT;
   }
 
-  if (superClass.type !== 'Identifier') {
-    throw translationError(
-      superClass,
-      `SuperClass: Non identifier super type of "${superClass.type}" not supported`,
-      context,
-    );
+  switch (superClass.type) {
+    case 'Identifier':
+    case 'MemberExpression': {
```

Comment:
The member expression case needs to be converted to a `QualifiedTypeIdentifier`.